### PR TITLE
Task reminder

### PR DESCRIPTION
**Pull request by M-Muaz-Shoukat**

### CODE REVIEW

File: todo_list_app/urls.py [hunk 1]
```diff
@@ -5,12 +5,10 @@
 router = DefaultRouter(trailing_slash=False)
 
 router.register(r'categories', views.CategoryViewSet, basename='category')
+router.register(r'tasks', views.TaskViewSet, basename='task')
```
Comment (m-shayan-ahmad) [inline todo_list_app/urls.py:8]: What is the purpose of `r` here?

File: todo_list_app/views.py [hunk 2]
```diff
@@ -107,20 +104,23 @@ def post(self, request):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
-def index(request):
-    return render(request, 'todo_list_app/index.html')
+class IsOwnerCategoryPermission(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+
+        return obj.user == request.user
```
Comment (m-shayan-ahmad) [inline todo_list_app/views.py:111]: Try to find out a generic solution that can be used for any Model.

File: todo_list_app/views.py [hunk 3]
```diff
@@ -130,110 +130,29 @@ def perform_create(self, serializer):
...
+
+    def get_queryset(self):
+        query = self.request.query_params.get('q')
+        task_list = Task.objects.filter(category__user=self.request.user)
 
         if query:
             task_list = task_list.filter(
                 Q(title__icontains=query) |
                 Q(description__icontains=query) |
                 Q(category__name__icontains=query)
```
Comment (m-shayan-ahmad) [inline todo_list_app/views.py:145]: What is the exact purpose of `icontaints` and specifically `i` in it?

**Review by m-shayan-ahmad — APPROVED** [review]

File: todo_list_app/urls.py (hunk 1 above)
Comment (M-Muaz-Shoukat) [reply todo_list_app/urls.py:8]: By default string takes \ -> backslash as escape character for exmaple if you want to add " -> double quotes in string the interpreter may take it as end of string and gives error on further characters e.g "hello "world"" so we '\' to tackle this "hello \"world\""

'r' at the front of string mark it as raw string in order to take '\' as character not as escape character

File: todo_list_app/views.py (hunk 3 above)
Comment (M-Muaz-Shoukat) [reply todo_list_app/views.py:145]: contains is used to perform case-sensitive search based on the value and icontains perform case-insensitive search

**Pull request merged**